cue/context_test.go

```diff
@@ -80,3 +84,58 @@ func TestNewList(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildInstancesSuccess(t *testing.T) {
```

Comment:
I personally would use a table-driven test here. But it seems in general that such tests could benefit from some common infrastructure using txtar files.

This can be done in a followup CL.